app/api/api_handlers.py

```diff
@@ -0,0 +1,50 @@
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel
+from app.models import database
+
+"""
+This file hanldes the various API endpoints, i.e post, get
+
+The general flow of information is as follows:
+    When a shortlink must be created:
+    customer -> api_handlers -> url_shortener -> api_handlers -> database.py -> api_handlers ->  customer
```

Comment:
A better representation would be

customer <-> api <-> storage <-> database

Your diagram is true, but it's redundant since it repeats many parts. 

The assumption is that the url_shortener is deeper in the call stack or lower level than the api_handler. In other words, url_shortener is a dependency of api_handler. 

It's important to identify that the dependency graph is 1 way here. i.e. so there's no cycle.